src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Properties/VsLangProjectPropertiesTests.cs

```diff
@@ -0,0 +1,160 @@
+
+using System;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Moq;
+using VSLangProj;
+using VSLangProj110;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    [ProjectSystemTrait]
+    public class VsLangProjectPropertiesTests
+    {
+        [ProjectSystemTrait]
+        public class VsLangProjectPropertiesProviderTests
+        {
+            [Fact]
+            public void Constructor_NullAsVsProject_ThrowsArgumentNull()
+            {
+                Assert.Throws<ArgumentNullException>("vsProject", () =>
+                {
+                    GetVsLangProjectProperties();
+                });
+            }
+
+            [Fact]
+            public void Constructor_NullAsThreadingService_ThrowsArgumentNull()
+            {
+                Assert.Throws<ArgumentNullException>("threadingService", () =>
+                {
+                    GetVsLangProjectProperties(Mock.Of<VSProject>());
+                });
+            }
+
+            [Fact]
+            public void Constructor_NullAsProjectProperties_ThrowsArgumentNull()
+            {
+                Assert.Throws<ArgumentNullException>("projectProperties", () =>
+                {
+                    GetVsLangProjectProperties(Mock.Of<VSProject>(), Mock.Of<IProjectThreadingService>());
+                });
+            }
+
+            [Fact]
+            public void VsLangProjectProperties_NotNull()
+            {
+                var provider = GetVsLangProjectProperties(Mock.Of<VSProject>(), Mock.Of<IProjectThreadingService>(), Mock.Of<ActiveConfiguredProject<ProjectProperties>>());
+                Assert.NotNull(provider);
+            }
+
+            [Fact]
+            public void VsLangProjectProperties_OutputTypeEx_Get()
+            {
+                var project = UnconfiguredProjectFactory.Create();
+                var enumValue = new Mock<IEnumValue>();
+                enumValue.Setup(s => s.DisplayName).Returns("2");
+                var data = new PropertyPageData()
+                {
+                    Category = ConfigurationGeneralBrowseObject.SchemaName,
+                    PropertyName = ConfigurationGeneralBrowseObject.OutputTypeExProperty,
+                    Value = enumValue.Object,
+                };
+
+                var projectProperties = ProjectPropertiesFactory.Create(project, data);
+                var activeConfiguredProject = ActiveConfiguredProjectFactory.ImplementValue(() => projectProperties);
+
+                var vsLangProjectProperties = new VsLangProjectProperties(Mock.Of<VSProject>(), IProjectThreadingServiceFactory.Create(), activeConfiguredProject);
+                Assert.Equal(vsLangProjectProperties.OutputTypeEx, prjOutputTypeEx.prjOutputTypeEx_Library);
+            }
+
+            [Fact]
+            public void VsLangProjectProperties_OutputType_Get()
```

Comment:
I was not able to write the tests for set methods because of Moq. I tried the same test below and added the following code but it does not seem to work.

```C#
var mockObject = Mock.Get(projectProperties.GetConfigurationGeneralBrowseObjectPropertiesAsync().Result.OutputType); // Get the mock object of IProperty, which is [here](https://github.com/dotnet/roslyn-project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/ProjectPropertiesFactory.cs#L110)
mockObject.Verify(m => m.SetValueAsync(It.IsAny<object>()));
```

Is this valid? Am I missing something here?